packages/hardhat/contracts/Ethernauts.sol

```diff
@@ -13,35 +13,48 @@ contract Ethernauts is ERC721Enumerable, Ownable {
     using Strings for uint256;
 
     // Can be set only once on deploy
-    uint public immutable maxTokens;
-    uint public immutable maxGiftable;
-    uint public immutable batchSize;
+    uint256 public immutable maxTokens;
+    uint256 public immutable maxGiftable;
+    uint256 public immutable batchSize;
 
-    // Can be changed by owner
+    // Can be changed by owner until minting stopped
     string public baseTokenURI;
-    uint public mintPrice;
-    uint public earlyMintPrice;
+
+    uint256 public mintPrice;
+
+    uint256 public earlyMintPrice;
+
     address public couponSigner;
 
     // Internal usage
-    uint private _tokensGifted;
+    uint256 private _tokensGifted;
     mapping(address => bool) private _redeemedCoupons; // user address => if its single coupon has been redeemed
-    uint[] private _randomNumbers;
+    uint256[] private _randomNumbers;
+
+    bool public permanentUrl;
 
     // Three different sale stages:
     enum SaleState {
         Paused, // No one can mint, except the owner via gifting (default)
         Early, // Only community can mint, at a discount using signed messages
         Open // Anyone can mint
     }
+
     SaleState public currentSaleState;
+    event SaleStateChanged(SaleState state);
+    event BaseTokenURIChanged(string baseTokenURI);
+    event EarlyMintPriceChanged(uint256 earlyMintPrice);
+    event MintPriceChanged(uint256 mintPrice);
+    event CouponSignerChanged(address couponSigner);
+    event WithdrawTriggered(address beneficiary);
+    event PermanentURITriggered(bool value);
```

Comment:
Is this the correct event name?

OpenSea uses `event PermanentURI(string _value, uint256 indexed _id);`
Ref: https://docs.opensea.io/docs/metadata-standards#freezing-metadata

Reply:
Yes, but opensea architecture of their 1155 storefront tokens is diff than ours

Reply:
so this does not apply to us. We need this event just to know when we will lock the changing of URLs once the minting is finished. Once it will be finished, we will be triggering this manually.